### PR TITLE
Make tests more stable

### PR DESCRIPTION
**Pull request by jerben**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,16 +1,41 @@
...
+	SIHL_ENV=test ./_build/default/sihl-storage/test/storage_mariadb.exe
+	SIHL_ENV=test ./_build/default/sihl-user/test/password_reset_mariadb.exe
+	SIHL_ENV=test ./_build/default/sihl-user/test/token_mariadb.exe
+	SIHL_ENV=test ./_build/default/sihl-user/test/user_mariadb.exe
+	SIHL_ENV=test ./_build/default/sihl-user/test/authn_mariadb.exe
+	SIHL_ENV=test ./_build/default/sihl-web/test/csrf_mariadb.exe
+	SIHL_ENV=test ./_build/default/sihl-web/test/session_mariadb.exe
+	SIHL_ENV=test ./_build/default/sihl-web/test/flash_mariadb.exe
+
+test-postgresql: build
```
Comment (aronerben) [inline Makefile:28]: Why are there fewer tests for postgres?

File: Makefile [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
 	opam exec -- dune build
 
 clean:
 	opam exec -- dune clean
 
-test:
+test-unit: build
 	SIHL_ENV=test opam exec -- dune test
 
-test-slow:
```
Comment (aronerben) [inline Makefile:12]: Please adjust CI too (CI still references `test-slow`)

**Review by aronerben — CHANGES_REQUESTED** [review]

**Comment by aronerben** [comment]
The fix looks good, thanks, but I still think we should be able to resolve the naming conflict `dune test` regarding `opam` deployment somehow without patching around it. Long-term maybe the possibility should come from `opam`.

File: Makefile (hunk 2 above)
Comment (jerben) [reply Makefile:12]: Fixed.

File: Makefile (hunk 1 above)
Comment (jerben) [reply Makefile:28]: Because for now we are not using the postgres backends, for some services I have implemented just the backends we currently use.

Comment (aronerben) [reply Makefile:28]: Ok, makes sense, thanks!

**Review by aronerben — APPROVED** [review]

**Pull request merged**